### PR TITLE
Add crash reason handle crash group

### PR DESCRIPTION
**Pull request by DominikWolek**

### CODE REVIEW

**Review by FelonEkonom — CHANGES_REQUESTED** [review]

File: lib/membrane/core/bin/callback_context.ex [hunk 1]
```diff
@@ -5,6 +5,7 @@ defmodule Membrane.Core.Bin.CallbackContext do
           [pad_options: map()]
           | [members: [Membrane.Child.name()], crash_initiator: Membrane.Child.name()]
```
Comment (FelonEkonom) [inline lib/membrane/core/bin/callback_context.ex:6]:
```suggestion
          | [members: [Membrane.Child.name()], crash_initiator: Membrane.Child.name(), reason: any()]
```

File: lib/membrane/core/bin/callback_context.ex [hunk 2]
```diff
@@ -5,6 +5,7 @@ defmodule Membrane.Core.Bin.CallbackContext do
...
           | [start_of_stream_received?: boolean()]
+          | [reason: any()]
```
Comment (FelonEkonom) [inline lib/membrane/core/bin/callback_context.ex:8]:
```suggestion
```

Comment (FelonEkonom) [reply lib/membrane/core/bin/callback_context.ex:8]: Fix this in the pipeline module as well

**Review by FelonEkonom — CHANGES_REQUESTED** [review]

File: lib/membrane/bin/callback_context.ex [hunk 3]
```diff
@@ -27,6 +27,7 @@ defmodule Membrane.Bin.CallbackContext do
           optional(:pad_options) => map(),
           optional(:members) => [Membrane.Child.name()],
           optional(:crash_initiator) => Membrane.Child.name(),
+          optional(:reason) => any(),
```
Comment (mat-hek) [inline lib/membrane/bin/callback_context.ex:30]: - let's call it `crash_reason`, so it's less ambiguous
- the type could be  `:normal | :shutdown | {:shutdown, term()} | term()` to be more descriptive, like [here](https://hexdocs.pm/elixir/GenServer.html#c:terminate/2)

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: test/support/child_crash_test/pipeline.ex [hunk 4]
```diff
@@ -92,4 +108,9 @@ defmodule Membrane.Support.ChildCrashTest.Pipeline do
 
     send(pid, {:create_path, spec})
   end
+
+  @spec inform_about_details_in_case_of_crash(pid(), pid()) :: any()
+  def inform_about_details_in_case_of_crash(pid, send_to) do
+    send(pid, {:inform_about_crash, send_to})
```
Comment (FelonEkonom) [inline test/support/child_crash_test/pipeline.ex:114]:
```suggestion
    send(pipeline, {:inform_about_crash, self()})
```

The only value, that is passed as `send_to`, is `self()`

File: test/membrane/integration/child_crash_test.exs [hunk 5]
```diff
@@ -67,6 +67,22 @@ defmodule Membrane.Integration.ChildCrashTest do
...
+
+    ChildCrashTest.Pipeline.add_path(pipeline_pid, [], :source, 1, :group_1)
+
+    [source_pid] = [:source] |> Enum.map(&get_pid_and_link(&1, pipeline_pid))
+
+    ChildCrashTest.Pipeline.inform_about_details_in_case_of_crash(pipeline_pid, self())
+
+    Process.exit(source_pid, :crash)
+
+    assert_receive({:crash, reason: :crash})
```
Comment (FelonEkonom) [inline test/membrane/integration/child_crash_test.exs:83]:
```suggestion
pipeline = Testing.Pipeline.start_supervised!(module: ChildCrashTest.Pipeline)
ChildCrashTest.Pipeline.add_path(pipeline_pid, [], :source, 1, :group_1)

# time for pipeline to start :source
Process.sleep(100)

ChildCrashTest.Pipeline.inform_about_details_in_case_of_crash(pipeline)

Testing.Pipeline.get_child_pid!(pipeline, :source)
|> Process.exit(:custom_crash_reason)

assert_receive {:crash, reason: :custom_crash_reason} 

Testing.Pipeline.terminate(pipeline)
```

1) `Process.flag(:trap_exit, true)` is unnecessary, we can just omit linking with pipeline and source.
2) It is necessary to terminate the pipeline at the end of the test

File: test/membrane/integration/child_crash_test.exs [hunk 6]
```diff
@@ -67,6 +67,22 @@ defmodule Membrane.Integration.ChildCrashTest do
     assert_pipeline_crash_group_down(pipeline_pid, 1)
   end
 
+  test "Pipeline receive correct crash reason" do
```
Comment (FelonEkonom) [inline test/membrane/integration/child_crash_test.exs:70]:
```suggestion
  test "Pipeline receives correct crash reason" do
```

**Review by FelonEkonom — CHANGES_REQUESTED** [review]

File: lib/membrane/core/parent/crash_group.ex [hunk 7]
```diff
@@ -15,9 +16,11 @@ defmodule Membrane.Core.Parent.CrashGroup do
...
           detonating?: boolean(),
-          crash_initiator: Membrane.Child.name()
+          crash_initiator: Membrane.Child.name(),
+          reason: :normal | :shutdown | {:shutdown, term()} | term()
         }
 
   @enforce_keys [:name, :mode]
-  defstruct @enforce_keys ++ [members: [], detonating?: false, crash_initiator: nil]
+  defstruct @enforce_keys ++
+              [members: [], detonating?: false, crash_initiator: nil, reason: nil]
```
Comment (FelonEkonom) [inline lib/membrane/core/parent/crash_group.ex:25]:
```suggestion
              [members: [], detonating?: false, crash_initiator: nil, crash_reason: nil]
```
Unify name of the field in `%CrashGroup{}` struct with name of the field passed in context to `handle_crash_group_down`

**Review by FelonEkonom — APPROVED** [review]

**Review by mat-hek — APPROVED** [review]

**Pull request merged**